### PR TITLE
PR: update de modelos de antecipação 

### PR DESCRIPTION
**Pull request by gmbarroso**
## Contexto
Este PR atualiza a forma como apresentamos os modelos de antecipação (1025  e D+X) na adição e nos detalhes do recebedor. Ele adiciona também um novo modelo _Customizado_ (modelo utilizado somente pela Magalu).

Ao inserir esses novos modelos o modal de ajuda teve que ser modificado. Então, este PR atualiza a forma como vemos o modal de Dúvidas também.

## Checklist
- [x] Atualização no cockpit
- [x] Atualização em Antecipação em detalhes do recebedor
- [x] Atualização no passo de conguração em adicionar recebedor
- [x] Atualização do modal de dúvidas

## Issues linkadas
- [Caesar #435](https://github.com/pagarme/caesar/issues/435)

## Screenshots
detalhes do recebedor:
image

Modal:
image

Adicionar recebedor
image

image

## Como testar?
- `git clone` desta repo
- `yarn` para instalar todas as dependências
- `yarn start` para rodar local
- Ir em _Recebedores_ e entrar no fluxo de _Adicionar Recebedor_. No passo de Configuração, testar todos os modelos. Fazer o mesmo em detalhes de qualquer recebedor. Quanto ao modal, testar se o mesmo está funcionando corretamente.

### CODE REVIEW

**Comment by rsmelo** [comment]
@gmbarroso  fiz o rebase da release/v0.21.0 agora de manha e precisa fazer o rebase na sua branch tambem =)

File: packages/pilot/src/validation/anticipationDays.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+const validateAnticipationDays = message => (value) => {
+  let parsed
+  try {
+    parsed = JSON.parse(`[${value}]`)
+  } catch (error) {
+    return message
+  }
+
+  const someInvalid = parsed.some((stringNumber) => {
```
Comment (vagnervst) [inline packages/pilot/src/validation/anticipationDays.js]: Seria bom um `ramda` maroto aqui 👀 

```js
const isBetween = (min, max) => both(
  gte(__, min),
  lte(__, max)
)

const areValidDays = all(
  allPass([
    is(Number), isBetween(1, 31)
  ])
)
```

File: packages/pilot/src/containers/RecipientDetails/Config/style.css [hunk 2]
```diff
@@ -30,11 +30,12 @@
...
 .imgSize {
-  width: 250px;
+  width: 450px;
 }
 
-.modalWidth {
-  width: 800px;
+.modalHeight {
+  overflow-y : auto;
+  height: 600px;
```
Comment (vagnervst) [inline packages/pilot/src/containers/RecipientDetails/Config/style.css]: Deveria mesmo ter esse `height` fixado? Pois gerou um scroll vertical no modal, quase ocultando o botão:

image

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: f6fc9f5f8aa58a57c76910e704e71aa7c0f546b5
  * lways use the standard `scope: action description`, sempre em inglês  
    - The **scope** shouldn't have spaces. If the word is composite, use dashes.  
    - The **action** should be **add**, **remove**, **refactor**, **move**,  **bump**, **update** or **release**  
    - The **description** should contain a quick description of the change

The complete commit style guide is available at https://github.com/pagarme/git-style-guide

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: f6fc9f5f8aa58a57c76910e704e71aa7c0f546b5
  * lways use the standard `scope: action description`, sempre em inglês  
    - The **scope** shouldn't have spaces. If the word is composite, use dashes.  
    - The **action** should be **add**, **remove**, **refactor**, **move**,  **bump**, **update** or **release**  
    - The **description** should contain a quick description of the change

The complete commit style guide is available at https://github.com/pagarme/git-style-guide

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: f6fc9f5f8aa58a57c76910e704e71aa7c0f546b5
  * lways use the standard `scope: action description`, sempre em inglês  
    - The **scope** shouldn't have spaces. If the word is composite, use dashes.  
    - The **action** should be **add**, **remove**, **refactor**, **move**,  **bump**, **update** or **release**  
    - The **description** should contain a quick description of the change

The complete commit style guide is available at https://github.com/pagarme/git-style-guide

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: f6fc9f5f8aa58a57c76910e704e71aa7c0f546b5
  * lways use the standard `scope: action description`, sempre em inglês  
    - The **scope** shouldn't have spaces. If the word is composite, use dashes.  
    - The **action** should be **add**, **remove**, **refactor**, **move**,  **bump**, **update** or **release**  
    - The **description** should contain a quick description of the change

The complete commit style guide is available at https://github.com/pagarme/git-style-guide

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: c823418002e64b03b3418d9d497fd51a33d89393
  * lways use the standard `scope: action description`, sempre em inglês  
    - The **scope** shouldn't have spaces. If the word is composite, use dashes.  
    - The **action** should be **add**, **remove**, **refactor**, **move**,  **bump**, **update** or **release**  
    - The **description** should contain a quick description of the change

The complete commit style guide is available at https://github.com/pagarme/git-style-guide

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: c823418002e64b03b3418d9d497fd51a33d89393
  * lways use the standard `scope: action description`, sempre em inglês  
    - The **scope** shouldn't have spaces. If the word is composite, use dashes.  
    - The **action** should be **add**, **remove**, **refactor**, **move**,  **bump**, **update** or **release**  
    - The **description** should contain a quick description of the change

The complete commit style guide is available at https://github.com/pagarme/git-style-guide

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: c823418002e64b03b3418d9d497fd51a33d89393
  * lways use the standard `scope: action description`, sempre em inglês  
    - The **scope** shouldn't have spaces. If the word is composite, use dashes.  
    - The **action** should be **add**, **remove**, **refactor**, **move**,  **bump**, **update** or **release**  
    - The **description** should contain a quick description of the change

The complete commit style guide is available at https://github.com/pagarme/git-style-guide

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: c823418002e64b03b3418d9d497fd51a33d89393
  * lways use the standard `scope: action description`, sempre em inglês  
    - The **scope** shouldn't have spaces. If the word is composite, use dashes.  
    - The **action** should be **add**, **remove**, **refactor**, **move**,  **bump**, **update** or **release**  
    - The **description** should contain a quick description of the change

The complete commit style guide is available at https://github.com/pagarme/git-style-guide

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 3a466c56fb3880b8eb62a0451096619765ea9256
  *   
￼ Good titles doesn't exceed 50 characters. To add more details use the body!  


 * Commit: 16dc20d12017c5640c7f4111a4fb6f6f9671f09b
  *   
￼ Good titles doesn't exceed 50 characters. To add more details use the body!

The complete commit style guide is available at https://github.com/pagarme/git-style-guide

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 3a466c56fb3880b8eb62a0451096619765ea9256
  *   
￼ Good titles doesn't exceed 50 characters. To add more details use the body!  


 * Commit: 16dc20d12017c5640c7f4111a4fb6f6f9671f09b
  *   
￼ Good titles doesn't exceed 50 characters. To add more details use the body!

The complete commit style guide is available at https://github.com/pagarme/git-style-guide

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 3a466c56fb3880b8eb62a0451096619765ea9256
  *   
￼ Good titles doesn't exceed 50 characters. To add more details use the body!  


 * Commit: 16dc20d12017c5640c7f4111a4fb6f6f9671f09b
  *   
￼ Good titles doesn't exceed 50 characters. To add more details use the body!

The complete commit style guide is available at https://github.com/pagarme/git-style-guide

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 3a466c56fb3880b8eb62a0451096619765ea9256
  *   
￼ Good titles doesn't exceed 50 characters. To add more details use the body!  


 * Commit: 16dc20d12017c5640c7f4111a4fb6f6f9671f09b
  *   
￼ Good titles doesn't exceed 50 characters. To add more details use the body!

The complete commit style guide is available at https://github.com/pagarme/git-style-guide

---
This message was auto-generated by https://gitcop.com

File: packages/cockpit/src/recipient/add/recipientBuilders.js [hunk 3]
```diff
@@ -2,50 +2,54 @@ import { range } from 'ramda'
...
+      recipientAnticipationData.automatic_anticipation_type = '1025'
+
+      break
+    case 'custom':
+      recipientAnticipationData.anticipatable_volume_percentage =
+        data.configuration.anticipationVolumePercentage ||
+        data.anticipationVolumePercentage
+      recipientAnticipationData.automatic_anticipation_enabled = true
+      recipientAnticipationData.automatic_anticipation_days =
+        `[${data.configuration.anticipationDays}]`
```
Comment (rsmelo) [inline packages/cockpit/src/recipient/add/recipientBuilders.js]: como ficaria o retorno desse caso com o  literal template?  o que valor é esperado possuir em `data.configuration.anticipationDays` nesse momento?

File: packages/pilot/public/locales/pt/translations.json [hunk 4]
```diff
@@ -492,12 +493,17 @@
...
       "anticipation_configuration": "Configurações de antecipação",
-      "anticipation_days": "Prazo de recebimento (dias)",
+      "anticipation_days": "Dias de Antecipação",
+      "anticipation_delay": "Delay (em dias)",
       "anticipation_model": "Modelo de antecipação",
       "anticipation_volume_percentage": "Volume antecipável (%)",
+      "anticipation_volume_percentage_1025_dx": "100",
       "anticipation_volume": "Volume antecipável",
       "automatic_1025": "Automática nos Dias 10/25",
+      "anticipation_1025_days": "10,25",
```
Comment (rsmelo) [inline packages/pilot/public/locales/pt/translations.json]: a label `anticipation_1025_days` e `anticipation_1025_delay` estão sendo usados como valores inciais do formulario.  Não deveriam estar sendo setadas em translations visto que são valores da regra de negocio e não devem ser alterados independente de idioma. O mais correto seria esse valor estar setado no componente que utiliza a label e não nas traduções

File: packages/pilot/src/containers/AddRecipient/ConfigurationStep/Anticipation/index.js [hunk 5]
```diff
@@ -39,17 +40,26 @@ const anticipationModelOptions = ({
     value: 'automatic_dx',
   }
 
+  const custom = {
+    name: 'Customizado',
```
Comment (rsmelo) [inline packages/pilot/src/containers/AddRecipient/ConfigurationStep/Anticipation/index.js]:
```suggestion
    name: t('pages.add_recipient.custom_anticipation'),
```


esse name deve vir de uma tradução assim como é feito nos demais items desse arquivo

File: packages/pilot/src/containers/AddRecipient/ConfigurationStep/Anticipation/index.js [hunk 6]
```diff
@@ -39,17 +40,26 @@ const anticipationModelOptions = ({
...
     availableOptions.push(automaticAtDays10And25)
+    availableOptions.push(automaticDX)
+    availableOptions.push(custom)
   }
 
   if (canConfigureAnticipation && maximumAnticipationDays >= 31) {
+    availableOptions.push(automaticAtDays10And25)
     availableOptions.push(automaticDX)
+    availableOptions.push(custom)
   }
```
Comment (rsmelo) [inline packages/pilot/src/containers/AddRecipient/ConfigurationStep/Anticipation/index.js]: o conteudo interno desses 2 `ifs` é igual, poderia ser feita uma validação que utilize dos dois
```suggestion
if (
!canConfigureAnticipation
|| (canConfigureAnticipation && maximumAnticipationDays >= 31)
) {
```

File: packages/pilot/src/validation/anticipationDays.js [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+)
+
+const areValidDays = all(
+  allPass([
+    is(Number), isBetween(1, 31),
+  ])
+)
+
+const validateAnticipationDays = message => (value) => {
+  let parsed
```
Comment (miglsoares) [inline packages/pilot/src/validation/anticipationDays.js]: Por que você declarou essa variável aqui? Acho que mutar ela no `try` me parece meio estranho.

File: packages/pilot/src/containers/AddRecipient/ConfigurationStep/Anticipation/index.js (hunk 5 above)
Comment (vagnervst) [reply packages/pilot/src/containers/AddRecipient/ConfigurationStep/Anticipation/index.js]: Notei que já até existe uma translation para `Customizado`, bastando utiliza-la:

`t('pages.add_recipient.custom_anticipation')`

File: packages/pilot/src/validation/anticipationDays.js [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+  gte,
+} from 'ramda'
+
+const isBetween = (min, max) => both(
+  gte(__, min),
+  lte(__, max)
+)
+
+const areValidDays = all(
+  allPass([
```
Comment (vagnervst) [inline packages/pilot/src/validation/anticipationDays.js]: Pensando melhor, aqui poderia usar `both` @matheusvellone, dado que são apenas duas condições validamos :slightly_smiling_face: Ficaria assim:

```js
all(
  both(
    is(Number),
    isBetween(1, 31)
  )
)
```

File: packages/cockpit/src/recipient/add/recipientBuilders.js [hunk 9]
```diff
@@ -2,50 +2,54 @@ import { range } from 'ramda'
 
 const getOnlyNumbers = string => string.replace(/\D/g, '')
 
-export const formatToAnticipation = (data, options = {}) => {
+export const formatToAnticipation = (data) => {
```
Comment (vagnervst) [inline packages/cockpit/src/recipient/add/recipientBuilders.js]: Se o `options` não é mais necessário nessa função, devemos remove-lo também de todos os lugares em que passamos ele:

https://github.com/pagarme/pilot/blob/a542b30f0d2a0b7e75b6df6906a025fc0873cd67/packages/cockpit/src/recipient/add/recipientBuilders.js#L115

https://github.com/pagarme/pilot/blob/a542b30f0d2a0b7e75b6df6906a025fc0873cd67/packages/cockpit/src/recipient/add/index.js#L3

https://github.com/pagarme/pilot/blob/a542b30f0d2a0b7e75b6df6906a025fc0873cd67/packages/pilot/src/pages/Recipients/Add/index.js#L79

**Comment by rsmelo** [comment]
ao selecionar o metodo de antecipação `Automatica nos dias 10/25` acontece o seguinte erro no console:
```js
Warning: Failed prop type: The prop `onChange` is marked as required in `Input`, but its value is `undefined`.```
![image](https://user-images.githubusercontent.com/479495/67695632-2566bc80-f984-11e9-97b3-b4043f2fd436.png)

File: packages/pilot/src/validation/anticipationDays.js (hunk 7 above)
Comment (Deivis) [reply packages/pilot/src/validation/anticipationDays.js]: Dado que há um padrão para o valor recebido, seria legal criar uma função que não use o `JSON.parse` para fazer esta validação, ex: 
```
pipe (
 split(', '),
 map(Number),
 reject(isNaN)
)
```
isso evitaria a necessidade do `try/catch`

**Comment by rsmelo** [comment]
@matheusvellone alterei a base do pr pra nova branch de release

File: packages/cockpit/src/recipient/add/recipientBuilders.js (hunk 9 above)
Comment (matheusvellone) [reply packages/cockpit/src/recipient/add/recipientBuilders.js]: Vou tirar os options desnecessários

File: packages/pilot/public/locales/pt/translations.json (hunk 4 above)
Comment (matheusvellone) [reply packages/pilot/public/locales/pt/translations.json]: Boa! Vi algumas outras chaves com o mesmo problema, aproveitei e corrigi elas também

File: packages/pilot/src/containers/AddRecipient/ConfigurationStep/Anticipation/index.js (hunk 5 above)
Comment (matheusvellone) [reply packages/pilot/src/containers/AddRecipient/ConfigurationStep/Anticipation/index.js]: Feito

File: packages/pilot/src/containers/AddRecipient/ConfigurationStep/Anticipation/index.js (hunk 6 above)
Comment (matheusvellone) [reply packages/pilot/src/containers/AddRecipient/ConfigurationStep/Anticipation/index.js]: Verdade!

File: packages/pilot/src/containers/AddRecipient/ConfirmStep/index.js [hunk 10]
```diff
@@ -174,24 +179,142 @@ const renderAnticipationConfig = (configuration, action, t) => {
...
+        && (
+        <Row>
+          <Col>
+            <span className={styles.infoTitle}>
+              {t('pages.add_recipient.anticipation_model')}
+            </span>
+            <span className={styles.info}>
+              {anticipationType}
+            </span>
+          </Col>
```
[the remark this answers is not in the document]
Comment (matheusvellone) [reply packages/pilot/src/containers/AddRecipient/ConfirmStep/index.js]: Pensei de criar uma função `renderSingleColumn` que retorna já algo como
```
<Col>
  <span className={styles.infoTitle}>
    {title}
  </span>
  <span className={styles.info}>
    {data}
  </span>
</Col>
```

O que acha @rsmelo ?

[the remark this answers is not in the document]
Comment (rsmelo) [reply packages/pilot/src/containers/AddRecipient/ConfirmStep/index.js]: boa, acho que é essa a ideia sim

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 08093a2f778993baac7abd852a4230b3c8b58dab
  * Always use the standard `scope: action description`, always in english  
    - The **scope** shouldn't have spaces. If the word is composite, use dashes.  
    - The **action** should be **add**, **remove**, **refactor**, **move**,  **bump**, **update** or **release**  
    - The **description** should contain a quick description of the change

The complete commit style guide is available at https://github.com/pagarme/git-style-guide

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 08093a2f778993baac7abd852a4230b3c8b58dab
  * Always use the standard `scope: action description`, always in english  
    - The **scope** shouldn't have spaces. If the word is composite, use dashes.  
    - The **action** should be **add**, **remove**, **refactor**, **move**,  **bump**, **update** or **release**  
    - The **description** should contain a quick description of the change

The complete commit style guide is available at https://github.com/pagarme/git-style-guide

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 08093a2f778993baac7abd852a4230b3c8b58dab
  * Always use the standard `scope: action description`, always in english  
    - The **scope** shouldn't have spaces. If the word is composite, use dashes.  
    - The **action** should be **add**, **remove**, **refactor**, **move**,  **bump**, **update** or **release**  
    - The **description** should contain a quick description of the change

The complete commit style guide is available at https://github.com/pagarme/git-style-guide

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 08093a2f778993baac7abd852a4230b3c8b58dab
  * Always use the standard `scope: action description`, always in english  
    - The **scope** shouldn't have spaces. If the word is composite, use dashes.  
    - The **action** should be **add**, **remove**, **refactor**, **move**,  **bump**, **update** or **release**  
    - The **description** should contain a quick description of the change

The complete commit style guide is available at https://github.com/pagarme/git-style-guide

---
This message was auto-generated by https://gitcop.com

File: packages/cockpit/src/recipient/add/recipientBuilders.js (hunk 3 above)
Comment (gmbarroso) [reply packages/cockpit/src/recipient/add/recipientBuilders.js]: @rsmelo O valor esperado a possuir no momento em que o usuário escolhe o modelo `custom` é o valor que ele entrar no input. O valor que ele entrar irá preencher um array, que é como o dado está sendo esperado. 

image

image


image

File: packages/pilot/src/containers/AddRecipient/ConfirmStep/index.js [hunk 11]
```diff
@@ -174,24 +179,142 @@ const renderAnticipationConfig = (configuration, action, t) => {
...
+          <Col>
+            <span className={styles.infoTitle}>
+              {t('pages.add_recipient.anticipation_days')}
```
[the remark this answers is not in the document]
Comment (gmbarroso) [reply packages/pilot/src/containers/AddRecipient/ConfirmStep/index.js]: Foi removido / alterado

File: packages/pilot/src/containers/AddRecipient/ConfirmStep/index.js [hunk 12]
```diff
@@ -174,24 +179,142 @@ const renderAnticipationConfig = (configuration, action, t) => {
...
+            <span className={styles.info}>
+              {anticipationDelay}
+            </span>
+          </Col>
+          <Col>
+            <span className={styles.infoTitle}>
+              {t('pages.add_recipient.anticipation_days')}
+            </span>
+            <span className={styles.info}>
+              {t('pages.add_recipient.anticipation_dx_days')}
```
[the remark this answers is not in the document]
Comment (gmbarroso) [reply packages/pilot/src/containers/AddRecipient/ConfirmStep/index.js]: Foi removido / alterado

File: packages/pilot/src/containers/AddRecipient/ConfirmStep/index.js [hunk 13]
```diff
@@ -174,24 +179,142 @@ const renderAnticipationConfig = (configuration, action, t) => {
...
+            <span className={styles.info}>
+              {anticipationType}
+            </span>
+          </Col>
+          <Col>
+            <span className={styles.infoTitle}>
+              {t('pages.add_recipient.anticipation_delay')}
+            </span>
+            <span className={styles.info}>
+              {t('pages.add_recipient.anticipation_1025_delay')}
```
[the remark this answers is not in the document]
Comment (gmbarroso) [reply packages/pilot/src/containers/AddRecipient/ConfirmStep/index.js]: Foi removido / alterado

File: packages/pilot/src/containers/AddRecipient/ConfirmStep/index.js [hunk 14]
```diff
@@ -174,24 +179,142 @@ const renderAnticipationConfig = (configuration, action, t) => {
...
+            </span>
+          </Col>
+          <Col>
+            <span className={styles.infoTitle}>
+              {t('pages.add_recipient.anticipation_days')}
+            </span>
+            <span className={styles.info}>
+              {t('pages.add_recipient.anticipation_1025_days')}
```
[the remark this answers is not in the document]
Comment (gmbarroso) [reply packages/pilot/src/containers/AddRecipient/ConfirmStep/index.js]: Foi removido / alterado

File: packages/pilot/src/containers/AddRecipient/ConfirmStep/index.js [hunk 15]
```diff
@@ -174,24 +179,142 @@ const renderAnticipationConfig = (configuration, action, t) => {
...
+            </span>
+          </Col>
+          <Col>
+            <span className={styles.infoTitle}>
+              {t('pages.add_recipient.anticipation_volume_percentage')}
+            </span>
+            <span className={styles.info}>
+              {t('pages.add_recipient.anticipation_volume_percentage_1025_dx')}
```
[the remark this answers is not in the document]
Comment (gmbarroso) [reply packages/pilot/src/containers/AddRecipient/ConfirmStep/index.js]: Foi removido / alterado

File: packages/pilot/src/containers/AddRecipient/ConfirmStep/index.js [hunk 16]
```diff
@@ -174,24 +179,142 @@ const renderAnticipationConfig = (configuration, action, t) => {
...
+            <span className={styles.info}>
+              {anticipationDays}
+            </span>
+          </Col>
+          <Col>
+            <span className={styles.infoTitle}>
+              {t('pages.add_recipient.anticipation_volume_percentage')}
+            </span>
+            <span className={styles.info}>
+              {t('pages.add_recipient.anticipation_volume_percentage_1025_dx')}
```
[the remark this answers is not in the document]
Comment (gmbarroso) [reply packages/pilot/src/containers/AddRecipient/ConfirmStep/index.js]: Foi removido / alterado

File: packages/pilot/src/validation/anticipationDays.js (hunk 8 above)
Comment (gmbarroso) [reply packages/pilot/src/validation/anticipationDays.js]: Feito!

File: packages/pilot/src/validation/anticipationDays.js (hunk 7 above)
Comment (gmbarroso) [reply packages/pilot/src/validation/anticipationDays.js]: Feito, alterado.

File: packages/cockpit/src/recipient/detail/format/formatRecipient.js [hunk 17]
```diff
@@ -24,25 +24,33 @@ export function formatHeaderData (data) {
 
 export function formatAnticipationData (data) {
   const anticipation = {
-    anticipationDays: data.automatic_anticipation_days || '',
+    anticipationDays: JSON.parse(data.automatic_anticipation_days || '[]'),
     anticipationModel: '',
     anticipationVolumePercentage: data
       .anticipatable_volume_percentage.toString(),
+    anticipationDelay: data.automatic_anticipation_1025_delay.toString() || '',
```
Comment (Deivis) [inline packages/cockpit/src/recipient/detail/format/formatRecipient.js]: Por quê aqui está sendo usado um `.toString` e na linha 27 um `JSON.parse` ? 
Poderia ser padronizado isso não ?

Comment (vagnervst) [reply packages/cockpit/src/recipient/detail/format/formatRecipient.js]: Concordo, porém o `JSON.parse` me parece necessário para termos um array a partir do que vem da API (acredito que respondemos `anticipation_days` como string).

Mas notei que na linha 52 estamos fazendo um `join` por virgula neste array. Daria pra simplificar isto com um `toString` logo depois do `JSON.parse` :thinking:

File: packages/pilot/src/containers/AddRecipient/ConfigurationStep/index.js [hunk 18]
```diff
@@ -142,7 +148,8 @@ class ConfigurationsStep extends Component {
           onSubmit={this.onFormSubmit}
           validateOn="blur"
           validation={{
-            anticipationDays: [required, isNumber, atLeastMinimumDays],
+            anticipationDays: [required, validateAnticipationDays],
+            anticipationDelay: [required, isNumber, atLeastMinimumDays],
```
Comment (vagnervst) [inline packages/pilot/src/containers/AddRecipient/ConfigurationStep/index.js]: Agora que estamos usando a validação `atLeastMinimumDays` em `anticipationDelay`, acho que deveríamos mudar o nome da variável

File: packages/cockpit/src/recipient/detail/format/formatRecipient.js (hunk 17 above)
Comment (gmbarroso) [reply packages/cockpit/src/recipient/detail/format/formatRecipient.js]: @Deivis Precisamos do `JSON.parse` para termos um array. Então não consigo tirar ele. Precisamos do `toString` pq precisamos passar o que vem para string pois é como o componente está esperando. Caso contrário, teremos warnings na aplicação.

@vagnervst No final das contas, a sua sugestão dá no mesmo. Só que nós temos um motivo para não fazer esse `toString` no início, logo depois do `JSON.parse`: dá uma olhada na linha 43. Se eu transformar tudo para string, terei que medir o tamanho da string. E fazer isso é mais complicado do que da forma que já está implementado.

Do jeito que está agora eu meço o tamanho do array através da quantidade de elementos dentro deles. Se eu transformo para string terei que medir a quantidade de caracteres e para isso as vírgulas terão que ser contadas tb. Não acho esse approach o correto. Vou deixar esse pedaço de código do jeito que está.

File: packages/pilot/src/containers/RecipientDetails/Config/index.js [hunk 19]
```diff
@@ -200,6 +202,15 @@ class RecipientDetailConfig extends Component {
         </div>
       )
     }
+    if (anticipation.anticipationModel === 'custom') {
```
Comment (vagnervst) [inline packages/pilot/src/containers/RecipientDetails/Config/index.js]: @gmbarroso Consegui simplificar bastante esses `if` com uma função `cond` do `ramda`:

```js
const getAnticipationModelTranslation = t => cond([
  [equals('manual'), always(t('pages.add_recipient.manual_volume'))],
  [equals('automatic_volume'), always(t('pages.add_recipient.automatic_volume'))],
  [equals('automatic_1025'), always(t('pages.add_recipient.automatic_1025'))],
  [equals('automatic_dx'), always(t('pages.add_recipient.automatic_dx'))],
  [equals('custom'), always(t('pages.add_recipient.custom_anticipation'))],
])
```

E então basta usar ela no `renderAnticipationSub` no lugar desses `if`:

```jsx
const getAnticipationModelName = getAnticipationModelTranslation(t)
const anticipationModelName = getAnticipationModelName(
  anticipation.anticipationModel
)

const HelpButton = (
  <Button
    type="button"
    size="tiny"
    fill="outline"
    onClick={this.handleOpenHelpModal}
  >
    {t('pages.recipient_detail.help')}
  </Button>
)

return (
  <div className={style.alignItems}>
    {`${model}: ${anticipationModelName}`}
    { anticipation.anticipationVolumePercentage && (
      <Fragment>
        <Spacing size="large" />
        {`${volume}: ${anticipation.anticipationVolumePercentage}%`}
      </Fragment>
    )}
    <Spacing size="large" />
    {HelpButton}
  </div>
)
```

Simplifiquei também o JSX pra não precisar ter mais de um no código. O que acha?

File: packages/cockpit/src/recipient/detail/format/formatRecipient.js (hunk 17 above)
Comment (vagnervst) [reply packages/cockpit/src/recipient/detail/format/formatRecipient.js]: @gmbarroso Faz sentido! Não havia visto o uso dos `.length` ali

File: packages/pilot/src/containers/AddRecipient/ConfigurationStep/index.js (hunk 18 above)
Comment (gmbarroso) [reply packages/pilot/src/containers/AddRecipient/ConfigurationStep/index.js]: Feito. Modifiquei o nome para `atLeastMinimumDelay`.

File: packages/pilot/src/containers/RecipientDetails/Config/index.js (hunk 19 above)
Comment (gmbarroso) [reply packages/pilot/src/containers/RecipientDetails/Config/index.js]: @vagnervst Gostei dessa abordagem. Eu modifiquei, só fiz um pouco diferente. Mantive as traduções nas variáveis.

File: packages/pilot/src/containers/AddRecipient/ConfigurationStep/Anticipation/index.js [hunk 20]
```diff
@@ -62,27 +70,108 @@ const renderAnticipationInput = (data, t) => {
...
   )
 
-  let inputProps = {
-    label: t('pages.add_recipient.anticipation_days'),
-    name: 'anticipationDays',
-  }
+  let inputProps
 
-  if (volumePercentage(anticipationModel)) {
+  if (volumePercentage(anticipationModel) || anticipationModel === 'custom') {
```
Comment (Deivis) [inline packages/pilot/src/containers/AddRecipient/ConfigurationStep/Anticipation/index.js]: Esta função `volumePercentage` havia me passado batido, ela pode ser escrita de uma maneira mais simples de ler ex:
```suggestion
  if (isVolumePercentage(anticipationModel) || anticipationModel === 'custom') {
```

File: packages/pilot/src/containers/RecipientDetails/Config/index.js (hunk 19 above)
Comment (vagnervst) [reply packages/pilot/src/containers/RecipientDetails/Config/index.js]: Boa @gmbarroso, gostei da forma como fez

File: packages/pilot/src/containers/AddRecipient/ConfirmStep/index.js [hunk 21]
```diff
@@ -174,24 +162,86 @@ const renderAnticipationConfig = (configuration, action, t) => {
...
+      }
+      {anticipationModel === 'custom'
+        && (
+        <Row>
+          {renderSingleColumn(
+            t('pages.add_recipient.anticipation_model'),
+            anticipationType
+          )}
+          {renderSingleColumn(
+            t('pages.add_recipient.anticipation_days'),
```
Comment (vagnervst) [inline packages/pilot/src/containers/AddRecipient/ConfirmStep/index.js]: A label de `anticipation_days` aqui está repetida:

image

File: packages/cockpit/src/recipient/add/recipientBuilders.js [hunk 22]
```diff
@@ -2,50 +2,54 @@ import { range } from 'ramda'
...
-        recipientAnticipationData.automatic_anticipation_type = '1025'
-      }
       break
     case 'automatic_volume':
       recipientAnticipationData.anticipatable_volume_percentage =
         data.configuration.anticipationVolumePercentage
       recipientAnticipationData.automatic_anticipation_enabled = true
+      recipientAnticipationData.automatic_anticipation_type = 'full'
 
-      if (canConfigureAnticipation) {
```
Comment (vagnervst) [inline packages/cockpit/src/recipient/add/recipientBuilders.js]: É correto retirar essa verificação? Pois agora sempre mandamos o `automatic_anticipation_type` e não consigo criar recebedores na minha company

File: packages/pilot/src/containers/AddRecipient/ConfigurationStep/Anticipation/index.js (hunk 20 above)
Comment (gmbarroso) [reply packages/pilot/src/containers/AddRecipient/ConfigurationStep/Anticipation/index.js]: Atualizado. Faz sentido esse nome :+1:

File: packages/cockpit/src/recipient/add/recipientBuilders.js (hunk 22 above)
Comment (matheusvellone) [reply packages/cockpit/src/recipient/add/recipientBuilders.js]: Que erro tá dando @vagnervst ?
Pq se vc não tem permissão pra criar recebedor com essa configuração, vc não deveria nem estar conseguindo selecionar a opção para criar recebedores com essa configuração

Comment (vagnervst) [reply packages/cockpit/src/recipient/add/recipientBuilders.js]: @matheusvellone Tomo um `403` com a mensagem `this company doesn't have permission to change the anticipation parameters from default values`. Porém se não envio os parâmetros:
- automatic_anticipation_enabled
- automatic_anticipation_days
- automatic_anticipation_type

Consigo criar o recebedor normalmente, é aí que está minha dúvida, se devemos enviar este parâmetro sempre (para qualquer um dos modelos de antecipação, como estamos fazendo) ou só quando possuo permissão :thinking:

File: packages/pilot/src/containers/AddRecipient/ConfirmStep/index.js (hunk 21 above)
Comment (gmbarroso) [reply packages/pilot/src/containers/AddRecipient/ConfirmStep/index.js]: @vagnervst Vagnão, aparentemente resolvido. Testar quando puder, por favor.

File: packages/cockpit/src/recipient/add/recipientBuilders.js (hunk 22 above)
Comment (gmbarroso) [reply packages/cockpit/src/recipient/add/recipientBuilders.js]: Feito!

**Pull request merged**